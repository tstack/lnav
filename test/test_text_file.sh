#! /bin/bash

export YES_COLOR=1
unset XDG_CONFIG_HOME

run_cap_test ${lnav_test} -n \
    ${top_srcdir}/README.md

run_cap_test ${lnav_test} -n -c ':goto #screenshot' \
    ${top_srcdir}/README.md

run_cap_test ${lnav_test} -n ${top_srcdir}/README.md#screenshot

# run_cap_test ${lnav_test} -n ${test_dir}/non-existent:4

run_cap_test ${lnav_test} -n ${top_srcdir}/README.md:-4

run_cap_test ${lnav_test} -n \
    -c ':goto 115' \
    -c ";SELECT top_meta FROM lnav_views WHERE name = 'text'" \
    -c ':write-json-to -' \
    ${top_srcdir}/README.md

run_cap_test ${lnav_test} -n \
    ${top_srcdir}/src/log_level.cc

cp ${test_dir}/UTF-8-test.txt UTF-8-test.md
run_cap_test ${lnav_test} -n \
    UTF-8-test.md

run_cap_test ${lnav_test} -n \
    -c ';SELECT * FROM lnav_file_metadata' \
    ${test_dir}/textfile_0.md
