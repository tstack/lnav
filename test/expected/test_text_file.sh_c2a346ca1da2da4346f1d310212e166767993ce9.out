[
    {
        "top_meta": {
            "file": "{top_srcdir}/README.md",
            "anchor": "#usage-with-systemd-journald-",
            "breadcrumbs": [
                {
                    "display_value": "README.md",
                    "search_placeholder": "",
                    "possibilities": [
                        {
                            "display_value": "README.md"
                        }
                    ]
                },
                {
                    "display_value": "LNAV – The Logfile Navigator",
                    "search_placeholder": "",
                    "possibilities": [
                        {
                            "display_value": "LNAV – The Logfile Navigator"
                        }
                    ]
                },
                {
                    "display_value": "Usage",
                    "search_placeholder": "",
                    "possibilities": [
                        {
                            "display_value": "Contributing"
                        },
                        {
                            "display_value": "Installation"
                        },
                        {
                            "display_value": "Links"
                        },
                        {
                            "display_value": "Screenshot"
                        },
                        {
                            "display_value": "See Also"
                        },
                        {
                            "display_value": "Support"
                        },
                        {
                            "display_value": "Usage"
                        }
                    ]
                },
                {
                    "display_value": "Usage with  systemd-journald ",
                    "search_placeholder": "",
                    "possibilities": [
                        {
                            "display_value": "Usage with  systemd-journald "
                        }
                    ]
                }
            ]
        }
    }
]
